Src/DependencyCollector/Shared.Tests/Implementation/DependencyCollectorDiagnosticListenerTests.Netstandard16.cs

```diff
@@ -103,6 +103,70 @@ public void OnRequestWithRequestEventWithNoRequestUri()
             Assert.AreEqual(0, this.sentTelemetry.Count);
         }
 
+        /// <summary>
+        /// Very second request does not throw an exception.
+        /// </summary>
+        [TestMethod]
+        public void VerifyOnRequestWithSameDoesNotThrowException()
+        {
+            Guid loggingRequestId = Guid.NewGuid();
+
+            var request = new HttpRequestMessage(HttpMethod.Get, RequestUrlWithScheme);
+
+            this.listener.OnRequest(request, loggingRequestId);
+            this.listener.OnRequest(request, loggingRequestId);
+        }
+
+        /// <summary>
+        /// Scenario: Assume a retry request after first request fails.
+        /// Very second request does not throw an exception.
+        /// Verify that telemetry is collected for both requests.
+        /// </summary>
+        [TestMethod]
+        public void VerifyOnRequestWithDuplicateRequestCreatesValidTelemetry()
+        {
+            Guid loggingRequestId = Guid.NewGuid();
+
+            var request = new HttpRequestMessage(HttpMethod.Get, RequestUrlWithScheme);
+
+            // first request, expected to fail
+            this.listener.OnRequest(request, loggingRequestId);
+
+            // second request (BEFORE HANDLING FIRST RESPONSE), expected to pass 
+            this.listener.OnRequest(request, loggingRequestId);
+
+            IOperationHolder<DependencyTelemetry> dependency;
+            Assert.IsTrue(this.listener.PendingDependencyTelemetry.TryGetValue(request, out dependency));
+            Assert.AreEqual(0, this.sentTelemetry.Count);
+
+            // first request fails
+            HttpResponseMessage response1 = new HttpResponseMessage(HttpStatusCode.NotFound)
+            {
+                RequestMessage = request
+            };
+
+            this.listener.OnResponse(response1, loggingRequestId);
+            Assert.IsFalse(this.listener.PendingDependencyTelemetry.TryGetValue(request, out dependency));
+            Assert.AreEqual(1, this.sentTelemetry.Count);
+            Assert.AreEqual(false, ((DependencyTelemetry)this.sentTelemetry.Last()).Success); // first request fails
+
+            Assert.Inconclusive();
```

Comment:
you can just split this into two separte unit tests, and [Ignore] the second one until fix is made. Easier to track [Ignore] tests as thats what we typically do to temp. disable tests.